openedx/features/course_experience/views/course_home.py

```diff
@@ -88,40 +88,56 @@ def render_to_fragment(self, request, course_id=None, **kwargs):
         Renders the course's home page as a fragment.
         """
         course_key = CourseKey.from_string(course_id)
+        course = get_course_with_access(request.user, 'load', course_key)
 
-        # Render the outline as a fragment
-        outline_fragment = CourseOutlineFragmentView().render_to_fragment(request, course_id=course_id, **kwargs)
+        # Render the course dates as a fragment
+        dates_fragment = CourseDatesFragmentView().render_to_fragment(request, course_id=course_id, **kwargs)
 
-        # Get resume course information
-        has_visited_course, resume_course_url = self._get_resume_course_info(request, course_id)
+        # Determine if the user is enrolled in the course (or is staff)
+        is_enrolled_or_staff = CourseEnrollment.is_enrolled(request.user, course_key)
```

Comment:
I got myself a bit confused here too. I think we want to allow course staff to access a course even if they are not enrolled, so this should be ``is_enrolled_or_staff``. If so, then I lost the check. I don't think we want ``is_enrolled`` to return True for staff that are not enrolled.